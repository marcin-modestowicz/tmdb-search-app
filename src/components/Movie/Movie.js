// @flow

import React, { Component } from "react";
import Image from "../Image";
import styles from "./Movie.css";

type Props = {
  movie: Object
};

const Movie = ({ movie }: Props) => {
  const {
    vote_average,
    title,
    poster_path,
    original_title,
    overview,
    release_date
  } = movie;
  const hasOriginalTitle = title !== original_title;
  const hasPoster = poster_path != null;

  return (
    <div className={styles.root}>
      <header className={styles.header}>
        <h3>{title}</h3>
        {hasOriginalTitle && <h4>{original_title}</h4>}
        <h5>
          Release date: <time>{release_date}</time>
        </h5>
        <h6>Rating: {vote_average}</h6>
      </header>
      <article className={styles.content}>
        {hasPoster && <Image name={poster_path} />}
        <p>{overview}</p>
      </article>
    </div>
  );
};

export default Movie;
