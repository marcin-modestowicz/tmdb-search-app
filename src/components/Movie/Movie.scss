.root {
  box-sizing: border-box;
  margin-top: 1em;
  padding: 1em 1em;
  width: 100%;
  max-width: 40em;
  font-family: sans-serif;

  .header {
    h3,
    h4,
    h5,
    h6 {
      margin-top: 0;
    }
  }

  .content {
    > img {
      float: left;
      margin: 0 1em 1em 0;
      width: 50%;
      max-width: 185px;
    }

    p {
      margin: 0;
      line-height: 1.5em;
    }
  }
}
