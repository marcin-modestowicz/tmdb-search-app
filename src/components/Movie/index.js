// @flow
export { default } from "./Movie";
