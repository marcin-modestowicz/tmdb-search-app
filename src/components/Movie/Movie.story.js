// @flow
import React from "react";
import { storiesOf } from "@storybook/react";
import Movie from "./Movie";

storiesOf("Movie", module).add("simple", () => (
  <Movie
    movie={{
      original_title: "Batman",
      overview:
        "The Dark Knight of Gotham City begins his war on crime with his first major enemy being the clownishly homicidal Joker, who has seized control of Gotham's underworld.",
      poster_path: "/kBf3g9crrADGMc2AMAMlLBgSm2h.jpg",
      release_date: "1989-06-23",
      title: "Batman",
      vote_average: 7
    }}
  />
));
