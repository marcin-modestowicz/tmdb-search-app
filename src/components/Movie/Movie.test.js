// @flow
import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Movie from "./Movie";

describe("Movie component", () => {
  test("should render properly", () => {
    const movieData = {
      original_title: "Batman",
      overview:
        "The Dark Knight of Gotham City begins his war on crime with his first major enemy being the clownishly homicidal Joker, who has seized control of Gotham's underworld.",
      poster_path: "/kBf3g9crrADGMc2AMAMlLBgSm2h.jpg",
      release_date: "1989-06-23",
      title: "Batman",
      vote_average: 7
    };
    const movie = shallow(<Movie movie={movieData} />);
    expect(toJson(movie)).toMatchSnapshot();
  });
});
