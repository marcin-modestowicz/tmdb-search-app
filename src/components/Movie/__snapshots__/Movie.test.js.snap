// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie component should render properly 1`] = `
<div
  className="root"
>
  <header
    className="header"
  >
    <h3>
      Batman
    </h3>
    <h5>
      Release date: 
      <time>
        1989-06-23
      </time>
    </h5>
    <h6>
      Rating: 
      7
    </h6>
  </header>
  <article
    className="content"
  >
    <Image
      name="/kBf3g9crrADGMc2AMAMlLBgSm2h.jpg"
    />
    <p>
      The Dark Knight of Gotham City begins his war on crime with his first major enemy being the clownishly homicidal Joker, who has seized control of Gotham's underworld.
    </p>
  </article>
</div>
`;
